k8s/upgrade/src/bin/upgrade-job/upgrade/utils.rs

```diff
@@ -209,3 +210,126 @@ pub(crate) async fn data_plane_is_upgraded(
 
     Ok(true)
 }
+
+/// Cordon storage node.
+pub(crate) async fn cordon_storage_node(
+    node_id: &str,
+    cordon_label: &str,
+    rest_client: &RestClientSet,
+) -> Result<()> {
+    let cordon_label = cordon_label.to_string();
+    loop {
```

Comment:
don't need the loops and explicit return anymore

Reply:
Removed the loops. Still a bunch of things can fail, maybe the returns should stay?